src/core/i18n/utils.js

```diff
@@ -230,6 +232,39 @@ function oneLineTranslationString(translationKey) {
   return translationKey;
 }
 
+type FormatFilesizeParams = {|
+  _filesize: typeof filesize,
+  i18n: I18nType,
+  size: number,
+|};
+
+// Translates a file size in bytes into a localized user-friendly format.
+export const formatFilesize = ({
+  _filesize = filesize,
+  i18n,
+  size,
+}: FormatFilesizeParams): string | null => {
+  const sizeStrings = {
+    // These are the expected values for the unit of measure returned by
+    // filesize. Realistically we shouldn't get anything back larger than TB.
+    B: i18n.gettext('%(formattedSize)s B'),
+    KB: i18n.gettext('%(formattedSize)s KB'),
+    MB: i18n.gettext('%(formattedSize)s MB'),
+    GB: i18n.gettext('%(formattedSize)s GB'),
+    TB: i18n.gettext('%(formattedSize)s TB'),
+  };
+
+  const [sizeNumber, sizeName] = _filesize(size).split(' ');
+  if (!sizeNumber || !sizeName) {
+    return null;
```

Comment:
Returning `null` seems unhelpful here. I would suggest doing this:

* Log an error so that Sentry will tell us we passed input that confused `filesize()`. If we see one of these for something out of our control, we could downgrade the log message to a warning.
* Return `i18n.formatNumber(sizeNumber)` without any units because I think that is a pretty good fallback. It may not make sense but it would make more sense than an empty string.